### PR TITLE
front: import timetalbe items e2e test

### LINKED ISSUE #13269: Tests e2e : import
### Description and goal

The test case: https://github.com/osrd-project/osrd-tests/issues/39

### Acceptance criteria

- [ ] Import a timetable item that contains both valid and invalid services and train schedules (covering all invalid cases), including various exceptions for the services
- [ ] Ensure that the imported train data has been processed correctly

### PR DESCRIPTION
**Pull request by Maymanaf**
closes https://github.com/OpenRailAssociation/osrd/issues/13269

### CODE REVIEW

**Comment by emersion** [comment]
Again, I don't believe an E2E test is the right tool for this job here. A unit test would be much better suited.

**Comment by Maymanaf** [comment]
> Again, I don't believe an E2E test is the right tool for this job here. A unit test would be much better suited.

I agree that unit tests are important here, but they don’t replace this E2E test ===> they complement it
E2E checks the real user flow in the browser (UI wiring, selectors, file upload, rendering) which unit tests can’t fully cover

File: front/tests/025-timetable-import.spec.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}>('public/locales/fr/operational-studies.json').main;
+
+const frCommonTranslations: CommonTranslations = readJsonFile('public/locales/fr/translation.json');
+const frTranslations = {
+  ...frTimetableTranslations,
+  ...frCommonTranslations,
+};
+
+test.describe('Verify timetable items import', () => {
+  test.use({ viewport: { width: 1920, height: 1080 } });
```
Comment (Synar) [inline front/tests/025-timetable-import.spec.ts]: Setting the viewport should no longer be needed since your last pr ^^

File: front/tests/pages/import-page.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    await expect(this.uploadFileDownloadButton).toBeVisible();
+    await this.uploadFileDownloadButton.click();
+
+    const resultsText = await this.importTimetableItemResults.innerText();
+
+    // Extract the numeric counts from the text
+    const [displayedPacedTrainCount, displayedTrainScheduleCount] =
+      resultsText.match(/\d+/g)?.map(Number) ?? [];
+
+    expect.soft(displayedPacedTrainCount).toBe(totalPacedTrainCount);
```
Comment (Synar) [inline front/tests/pages/import-page.ts]: I did not know about soft, it sounds nice. Although we should not abuse it, perhaps we should retrofit some older tests to use it?

(Outside of the scope of this pr obviously)

File: front/tests/pages/import-page.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  ): Promise<void> {
+    const { totalPacedTrainCount, totalTrainScheduleCount } = itemCounts;
+
+    await this.uploadFileDropzone.setInputFiles(filePath);
+    await expect(this.uploadFileDownloadButton).toBeEnabled();
+
+    await expect(this.uploadFileDownloadButton).toBeVisible();
+    await this.uploadFileDownloadButton.click();
+
+    const resultsText = await this.importTimetableItemResults.innerText();
```
Comment (Synar) [inline front/tests/pages/import-page.ts]: I'm afraid that the following expects could be flaky, so I wouldn't hate wrapping the result text verification in a poll preemptively. Feel free to ignore if you disagree.

File: front/tests/pages/operational-studies/timetable-items-details-section.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    await expect.soft(this.pacedTrainStopsCount.nth(pacedTrainIndex)).toHaveText(stopsCount);
+    await expect.soft(this.pacedTrainPathLength.nth(pacedTrainIndex)).toHaveText(pathLength);
+    await expect
+      .soft(this.pacedTrainEnergyConsumed.nth(pacedTrainIndex))
+      .toHaveText(energyConsumed);
+    await expect.soft(this.pacedTrainDurationTime.nth(pacedTrainIndex)).toHaveText(durationTime);
+  }
+
+  async trainScheduleLabels(trainScheduleIndex: number, expectedDetails: DetailRow): Promise<void> {
```
Comment (Synar) [inline front/tests/pages/operational-studies/timetable-items-details-section.ts]: The name of this function is a bit weird. By symmetry with verifyPacedTrainDetails, I would expect it to be called verifyTrainScheduleDetails, and the current verifyTrainScheduleDetails to be called verifyTrainSchedulesDetails or verifyAllTrainSchedulesDetails.

**Review by Synar — APPROVED** [review]
Overall code LGTM, thanks for the pr!

Could you tell me a bit more about the test data? If it was crafted with purpose (as in you intended certain schedules or paced trains to have certain behavior, features or edge cases), it would be nice to document this somewhere. Otherwise such purpose will inevitably be lost.

(Also no opinion on the unit vs e2e test debate.)

File: front/tests/025-timetable-import.spec.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+test.describe('Verify timetable items import', () => {
+  let importPage: ImportPage;
+  let timetableItemDetailSection: TimetableItemDetailSection;
+  let pacedTrainSection: PacedTrainSection;
+
+  let project: Project;
+  let study: Study;
+  let scenarioItems: Scenario;
```
Comment (clarani) [inline front/tests/025-timetable-import.spec.ts]:
```suggestion
  let scenario: Scenario;
```

File: front/tests/025-timetable-import.spec.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        './tests/assets/operation-studies/timetable-items.json',
+        EXPECTED_COUNTS
+      );
+      await importPage.launchTimetableItemImport(frTranslations.success);
+      await importPage.returnSimulationResult();
+    });
+
+    await test.step('Verify timetable items count after import', async () => {
+      await importPage.verifyTotalItemsLabel(frTranslations, EXPECTED_COUNTS);
+      await timetableItemDetailSection.showTimetableItemsDetails();
```
Comment (clarani) [inline front/tests/025-timetable-import.spec.ts]: This should be moved to the following step I think ?

File: front/tests/025-timetable-import.spec.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+      await timetableItemDetailSection.verifyPacedTrainDetails(0, PACED_DETAILS[0]);
+      await pacedTrainSection.verifyOccurrencesCount(3, 0);
+      await timetableItemDetailSection.verifyPacedTrainDetails(1, PACED_DETAILS[1]);
+      await pacedTrainSection.verifyOccurrencesCount(5, 3);
+    });
+    await test.step('Verify details for valid imported train schedules', async () => {
+      await timetableItemDetailSection.verifyTrainSchedulesDetails(TRAIN_SCHEDULE_DETAILS);
+    });
+    await test.step('Verify invalid items reasons', async () => {
+      await timetableItemDetailSection.verifyInvalidReasons([
```
Comment (clarani) [inline front/tests/025-timetable-import.spec.ts:96]: So here, you will have only 1 trainschedule with all these invalid reasons ?

**Review by clarani — COMMENTED** [review]
Here is a first batch of comments, but overall looks good !

File: front/tests/assets/constants/timetable-items-details.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  pathLength: string;
+  energyConsumed: string;
+  durationTime: string;
+};
+
+function itemStops(number: string): string {
+  return frScenarioTranslations.timetable.stopsCount_other.replace('{{count}}', number);
+}
+
+const itemStopSingular = frScenarioTranslations.timetable.stopsCount_one;
```
Comment (clarani) [inline front/tests/assets/constants/timetable-items-details.ts:23]: I think it would be easier to read and as efficient to simply write "x arrêts" in each `DetailRow`

File: front/tests/assets/constants/timetable-items-details.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+export const PACED_DETAILS: DetailRow[] = [
+  {
+    stopsCount: itemStops('2'),
```
Comment (clarani) [inline front/tests/assets/constants/timetable-items-details.ts:27]:
```suggestion
    stopsCount: "2 arrêts",
```

File: front/tests/assets/operation-studies/timetable-items.json
Comment (clarani) [inline front/tests/assets/operation-studies/timetable-items.json:1]: Should all trainsSchedules be added? If possible, try to reduce the size of the JSON file as much as possible to avoid overloading the Git repository.

File: front/tests/assets/operation-studies/timetable-items.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+---
+
+## TrainSchedules
+
+- Multiple rolling stock categories
+- Distinct itineraries using predefined trigrams or requested map points
+- Different margins and timetable offsets
+- Various input configurations
+- Diverse speed limit tags and labels
+- Includes intentionally invalid trains (missing path, invalid MR, wrong tag)
```
Comment (clarani) [inline front/tests/assets/operation-studies/timetable-items.md:17]: This diversity can be tested in the pacedTrains section

File: front/tests/025-timetable-import.spec.ts (hunk 7 above)
Comment (Maymanaf) [reply front/tests/025-timetable-import.spec.ts:96]: Here we have 3 trainshedules and 2 paced trains with the different invalid reasons

File: front/tests/025-timetable-import.spec.ts (hunk 5 above)
Comment (Maymanaf) [reply front/tests/025-timetable-import.spec.ts]: Done

File: front/tests/025-timetable-import.spec.ts (hunk 6 above)
Comment (Maymanaf) [reply front/tests/025-timetable-import.spec.ts]: Done

File: front/tests/assets/constants/timetable-items-details.ts (hunk 8 above)
Comment (Maymanaf) [reply front/tests/assets/constants/timetable-items-details.ts:23]: Yes, it’s easier to use "x arrêts", but we should keep the same logic since we lose the translation file. This helps keep the test maintainable and adaptable if we want to test different languages

File: front/tests/assets/operation-studies/timetable-items.md (hunk 10 above)
Comment (Maymanaf) [reply front/tests/assets/operation-studies/timetable-items.md:17]: it's applied in both trainSchedules and pacedTrains

File: front/tests/assets/operation-studies/timetable-items.json
Comment (Maymanaf) [reply front/tests/assets/operation-studies/timetable-items.json:1]: All the train schedules are different as explained in `timetable-items.md`, but I tried to reduce them to the minimum, so now we no longer cover all rolling stock categories

File: front/tests/assets/constants/timetable-items-details.ts (hunk 9 above)
Comment (Maymanaf) [reply front/tests/assets/constants/timetable-items-details.ts:27]: Same as above

**Review by clarani — APPROVED** [review]
